### PR TITLE
[2] Bill + Share Payment Screens

### PR DESCRIPTION
**Pull request by aivantg**
[//]: # "These comments meant for your reference, they are invisible and don't need to be deleted"

## What's new in this PR?

[//]: # '#####################################'
[//]: # '####### YOUR ANSWER BELOW ###########'

Cleaned up payment utilities and created Bill Payment page

[//]: # '############## END ##################'

### How to Review

[//]: # 'The order in which to review files and 
what to expect when testing locally'
[//]: # '#####################################'
[//]: # '####### YOUR ANSWER BELOW ###########'
[//]: # '############## END ##################'

### Related PRs

[//]: # "Optional - related PRs you're waiting on
/ PRs that will conflict, etc"
[//]: # '#####################################'
[//]: # '####### YOUR ANSWER BELOW ###########'

Second part of #118 

[//]: # '############## END ##################'

### Next steps

[//]: # "What doesn't work yet, what's NOT in this 
PR/has to be done "
[//]: # '#####################################'
[//]: # '####### YOUR ANSWER BELOW ###########'

- Styling Tune-Ups
- better loading states
- Need to figure out how and where to round to 2 decimal places
- need to figure out how to disable bill payment if bill is pending and show a better message
- need to add this to investments page

[//]: # '############## END ##################'

### Tests Performed, Edge Cases

[//]: # '#####################################'
[//]: # '####### YOUR ANSWER BELOW ###########'
[//]: # '############## END ##################'

### Screenshots

[//]: # '#####################################'
[//]: # '#### YOUR SCREENSHOTS BELOW ########'
[//]: # '############## END ##################'

CC: @dfangshuo

### CODE REVIEW

File: src/lib/subscriberUtils.js [hunk 1]
```diff
@@ -46,9 +46,22 @@ const getSubscriberTransactionData = async owner => {
...
+    activeBill = 0;
+  } else {
+    [activeBill] = activeBills;
+  }
+
+  // TODO: remove the number 2 on the server side
+  return {
+    activeBill: {
+      ...activeBill,
+      balance: Number(activeBill.balance.toFixed(2))
```
Comment (dfangshuo) [inline src/lib/subscriberUtils.js:61]: doesnt this mean that activeBill would have 2 balance fields? 1 for with the balance rounded off to do decimal places, and 1 with all decimal places?

Comment (dfangshuo) [reply src/lib/subscriberUtils.js:61]: also is this what you were talking about wrt to rounding?

File: src/screens/general/BuyShares.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { PayPalButton } from 'react-paypal-button-v2';
+import SharesProgressBar from './components/SharesProgressBar';
+import LeftArrow from '../../assets/left_arrow.png';
+import '../../styles/BuyShares.css';
+
+const clientId = process.env.REACT_APP_PAYPAL_CLIENT_ID;
+const SHARE_PRICE = 100;
```
Comment (dfangshuo) [inline src/screens/general/BuyShares.js:10]: move this into `constants.js`?

File: src/screens/general/BuyShares.js [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    super(props);
+    this.state = {
+      sharesBuying: 0
+    };
+  }
+
+  addShares = () => {
+    const { sharesBuying } = this.state;
+    const { owner } = this.props;
+    const maxAllowed = 10 - owner.numberOfShares;
```
Comment (dfangshuo) [inline src/screens/general/BuyShares.js:23]: Make 10 a constant (`MAX_ALLOWED_SHARES`) or something and move into `constants.js`

File: src/screens/general/components/SharesProgressBar.js [hunk 4]
```diff
@@ -17,14 +17,12 @@ export default class SharesProgressBar extends React.PureComponent {
...
     return (
-      <div className="circle-progress-bar">
-        <CircularProgressbar
-          viewBox="0 0 0 0"
-          value={percentage}
-          text={numberOfShares}
-          styles={buildStyles(circularProgressBarStyles)}
-        />
-      </div>
+      <CircularProgressbar
```
Comment (dfangshuo) [inline src/screens/general/components/SharesProgressBar.js:20]: was the initial class just redundant? or what is the reason for this change

File: src/screens/subscriber/components/PaymentButtons.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { PayPalButton } from 'react-paypal-button-v2/lib';
+
+const PaymentButtons = ({ amount }) => {
```
Comment (dfangshuo) [inline src/screens/subscriber/components/PaymentButtons.js:4]: What is this for? I just did a quick search in VSCODE and saw that this isn't being used anywhere?

**Comment by aivantg** [comment]
fun fact, I think this PR has the changes from Cindy's Shares PR strangely, which is why it's so big

will address changes and merge

File: src/lib/subscriberUtils.js (hunk 1 above)
Comment (aivantg) [reply src/lib/subscriberUtils.js:61]: yeah this is a hack honestly to make this work, but it would be ideal if this happened server side

File: src/screens/general/components/SharesProgressBar.js (hunk 4 above)
Comment (aivantg) [reply src/screens/general/components/SharesProgressBar.js:20]: this is from Cindy's PR

File: src/screens/subscriber/components/PaymentButtons.js (hunk 5 above)
Comment (aivantg) [reply src/screens/subscriber/components/PaymentButtons.js:4]: whoops, yeah let me delete

**Comment by aivantg** [comment]
Can affirm that bill payment works. Merging
